### DISCUSSION TITLE
ScriptReference/Debug

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#351] [Notes]
# ScriptReference/Debug

https://docs.unity3d.com/ScriptReference/Debug.html

**Comment by docscomments** [comment]
**Comment by `@jonHuffman`:**  

Keep in mind that the Debug Log methods do not get stripped in release builds and they can have a serious performance impact. Consider wrapping them in custom methods and making use of the [Conditional attribute](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.conditionalattribute) to disable them when desired.

**Comment by vertxxyz** [comment]
| Static Methods | |
| -- | -- |
| `DebugBreak` | Breaks into the attached debugger, if present. |
| `RetrieveStartupLogs` | Used to retrieve any logs captured before package startup. Logs are only captured when the Player Setting **Capture Startup Logs** has been set. |
